src/amo/components/LanguagePicker.js

```diff
@@ -1,14 +1,59 @@
+/* global window */
+import config from 'config';
 import React, { PropTypes } from 'react';
 import { compose } from 'redux';
+import { connect } from 'react-redux';
 
+import { languageJSON } from 'core/languages';
 import translate from 'core/i18n/translate';
+import { addQueryParams } from 'core/utils';
 
 import './LanguagePicker.scss';
 
 
+export function changeLocaleURL({ currentLocale, location, newLocale }) {
+  const newPath = location.pathname.replace(`/${currentLocale}/`,
```

Comment:
Unfortunately `split` doesn't return the remaining contents as one long string but instead just stops adding to the array it makes:

```node
> '/en-US/firefox/nowhere/'.split('/')
[ '', 'en-US', 'firefox', 'nowhere', '' ]
> '/en-US/firefox/nowhere/'.split('/', 2)
[ '', 'en-US' ]
> '/en-US/firefox/nowhere/'.split('/', 3)
[ '', 'en-US', 'firefox' ]
> '/en-US/firefox/nowhere/'.split('/', 1)
[ '' ]
> '/en-US/firefox/nowhere/'.split('/', 2)
[ '', 'en-US' ]
```

So using `split` would be more code and end up doing the same thing, especially as I realised we should only be matching the first `/locale/` in the URL (in the off chance an addon or other URL had `/locale/` in it we don't want to replace it).

So I've improved the Regex to only match the first instance and I think it's fine; the alternate with `split` involves more code that would go through the same process.